front-end/src/components/CustomInput.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect, useRef, useState } from "react"
+import { Markdown } from "@/components/Markdown.tsx"
+import { Input } from "@/components/ui/input"
+import { useFormContext } from "@/hooks/useFormContext.ts"
+import { useTranslation } from "../hooks/useTranslation"
+
+export const CustomInput: React.FC<{ id: string }> = ({ id }: { id: string }) => {
```

Comment:
rename this to FormInputField and move it to its own file under `@/components/ui/`. The component should be UI-only and not perform any logic from the Question (I think the only such logic left is disabled and invalid)